leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -39,29 +39,468 @@ var EvolvingFunctionBBM = Backbone.Model.extend({
+     *
+     * If funcType is RC, CR, MP, MN, SD, DS  a Constraint object
+     * representing an absolute constraint will be added. If not, this
+     * function does not do anything
+     *
+     * @param {String} funcType
+     *   ex: 'RC'
+     */
+    addAbsConstraint: function(funcType) {
+        if (funcType == 'RC' || funcType == 'CR' || funcType == 'MP' ||
+            funcType == 'MN' || funcType == 'SD' || funcType == 'DS') {
+            graph.constraints.push(new ConstraintBBM({type: 'A', srcID: this.get('id'), srcRefTP: 'A', destID: null, destRefTP: null}));
+        }
+    }, 
+ 
+ 
+    /**
+     * Returns the absolute time for this Intention's absolute constraint at
+     * the starting epoch boundary start
+     *
+     * @param {String} source
+     *  ex. 'A'
+     */
+    getAbsConstTime: function(source) {
+        return model.getAbsConstBySrcID(this.get('id'), source).absoluteValue;
+    }, 
+ 
+    /**
+     * Removes the absolute Constraint object(s) for this Intention from
+     * the global model variable, if such absolute Constraint object(s) exists
+     */
+
+    removeAbsConstraint: function() {
+        var i = 0;
+        while (i < graph.constraints.length) {
+            var constraint = graph.constraints.models[i];
+            if (constraint.get('type') == 'A' && constraint.get('srcID') === this.get('id')) {
+                graph.constraints.splice(i, 1);
+            } else {
+                i++;
+            }
+        }
+    }, 
+ 
+    /**
+     * Adds a new FuncSeg to the end of this Intention's EvolvingFunction's
+     * function list.
+ 
+     * Also adds a corresponding Constraint object reprenting an absolute constraint
+     * into the global model variable.
+     *
+     *This function should only be used to add new function
+     * segments for user defined functions
+     *
+     * @param {String} funcType
+     *   ex: 'C'
+     * @param {String} satValue
+     *   ex: '0000'
+     */
+    addUserDefinedSeg: function(funcType, satValue){
+ 
+        var len = this.evolvingFunction.getFuncSegments().length;
+        // originally used to be funcStop 
+        var start = this.evolvingFunction.getFuncSegments()[len - 1].get('stopTP');
+        var code = start.charCodeAt(0) + 1;
+        var stop = String.fromCharCode(code);
+ 
+        //create new funcsegment model 
+        var new_model = new FunctionSegmentBBM({type: funcType, refEvidencePair: satValue, startTP: start, stopTP: stop}); 
+        this.evolvingFunction.getFuncSegments().push(new_model); 
+ 
+        model.constraints.push(new ConstraintBBM({type: 'A', srcID: this.get('id'), srcRefTP: start, destID: null, destRefTP: null}));
+ 
+    }, 
+    /**
+     * Sets the marked value for the FuncSegments in the
+     * EvolvingFunction for this Intention
+     *
+     * This function will only be called for I, D, RC, MP, MN functions
+     */
+    setMarkedValueToFunction: function(satValue) {
```

Comment:
This will need to be changed/removed because we are getting rid of the associated drop down box.